### PR TITLE
Set request headers when request body is empty in Web Cmdlets

### LINKED ISSUE #9473: Invoke-RestMethod PUT fails
# Steps to reproduce

```powershell
 $Header = @{
Authorization = "Basic <MyToken>"
"Content-Type" =  "application/json"
}
Invoke-RestMethod -Uri https://MyOrg.freshservice.com/api/v2/assets/1511/delete_forever -headers $header -method PUT


```

# Expected behavior

```none

```
The API returns nothing when deleting an entry

# Actual behavior

```ps
Invoke-RestMethod : {"message":"Content-Type header should have application/json","code":"invalid_content_type"}
At line:1 char:1
+ Invoke-RestMethod @Params
+ ~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (Method: PUT, Reques\u2026PowerShell/6.2.0
}:HttpRequestMessage) [Invoke-RestMethod], HttpResponseException
+ FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
```

# Environment data

```ps
PS C:\> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.2.0
PSEdition                      Core
GitCommitId                    6.2.0
OS                             Microsoft Windows 10.0.18362
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
image

### PR DESCRIPTION
**Pull request by markekraus**
# PR Summary

Fixes #9473 by always setting an empty HttpContent when no body is present.

## PR Context

RE: #9473

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 1]
```diff
@@ -1241,26 +1241,29 @@ internal virtual void FillRequestStream(HttpRequestMessage request)
...
+                }
+                else
+                {
+                    try
                     {
-                        request.Content.Headers.TryAddWithoutValidation(entry.Key, entry.Value);
+                        request.Content.Headers.Add(entry.Key, entry.Value);
                     }
-                    else
+                    catch (FormatException ex)
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs]: It seems we haven't a test for the code branch.

Comment (markekraus) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs]: We do. This is not new code, just move outside of an if statement. 
example: https://github.com/PowerShell/PowerShell/blob/bf71b2c6161085ef12977fae88548b96cd93319d/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1#L944

**Review by iSazonov — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
@markekraus I tried to rewrite the PR title to be based on our guidance (describing the change)

**Comment by markekraus** [comment]
@TravisEz13 You actually had it reversed. should be good now.

**Comment by iSazonov** [comment]
@markekraus Thanks for your contribution.

**Pull request merged**